core/modules/system/system.admin.inc

```diff
@@ -285,31 +285,46 @@ function system_theme_enable() {
 }
 
 /**
- * Menu callback; Disables a theme.
+ * Theme disable confirmation form.
+ *
+ * @see system_theme_disable_confirm_submit()
+ *
+ * @ingroup forms
  */
-function system_theme_disable() {
+function system_theme_disable_confirm($form, &$form_state) {
   if (isset($_REQUEST['theme']) && isset($_REQUEST['token']) && backdrop_valid_token($_REQUEST['token'], 'system-theme-operation-link')) {
-    $theme = $_REQUEST['theme'];
-    // Get current list of themes.
-    $themes = list_themes();
+    $form['theme'] = $_REQUEST['theme'];
 
-    // Check if the specified theme is one recognized by the system.
-    if (!empty($themes[$theme])) {
-      if ($theme == config_get('system.core', 'theme_default')) {
-        // Don't disable the default theme.
-        backdrop_set_message(t('%theme is the default theme and cannot be disabled.', array('%theme' => $themes[$theme]->info['name'])), 'error');
-      }
-      else {
-        theme_disable(array($theme));
-        backdrop_set_message(t('The %theme theme has been disabled.', array('%theme' => $themes[$theme]->info['name'])));
-      }
+    $confirm_question = t('Are you sure you want to disable this theme? All custom configuration will be lost.');
```

Comment:
The "All custom configuration will be lost." bit should not be part of the page title/question. It should be a warning message instead:

```suggestion
    $confirm_question = t('Are you sure you want to disable this theme?');
    backdrop_set_message(t('All custom configuration will be lost!'), 'warning');
    backdrop_set_message(t('This action cannot be undone.'), 'warning');
```

Reply:
This PR uses the same format as content type deletion confirmation. I would suggest keeping it consistent with that one. And change them in all confirmation pages in a separate PR.